pkg/image/definition.go

```diff
@@ -28,6 +28,13 @@ type OperatingSystem struct {
 	Users      []OperatingSystemUser `yaml:"users"`
 	Systemd    Systemd               `yaml:"systemd"`
 	Suma       Suma                  `yaml:"suma"`
+	Packages   Packages              `yaml:"packages"`
+}
+
+type Packages struct {
+	PKGList  []string `yaml:"pkgList"`
+	AddRepos []string `yaml:"additionalRepos"`
```

Comment:
I was left with the impression that we should support a list of repositories as seen in zypper. Meaning either URLs, paths to local repositories or paths to local ".repo" files.

With the suggested implementation it should be possible to do all of this. So in theory you should be able to have something like:
```yaml
packages:
    additionalRepos:
    - https://download.nvidia.com/suse/sle15sp5/
    - /path/to/local/repo (relevant to the `eib-dir`)
    - /path/to/local/foo.repo (relevant to the `eib-dir`)
```
^ I have only tested the URL portion, but taking into account that we pass a split repo list directly to the [zypper command](https://github.com/ipetrov117/edge-image-builder/blob/rpm-resolution/pkg/repo/resolver/scripts/Dockerfile.tpl#L25) in the [Dockerfile](https://github.com/ipetrov117/edge-image-builder/blob/rpm-resolution/pkg/repo/resolver/scripts/Dockerfile.tpl#L23), I do not think adding paths to local dirs/".repo" files would be a problem (will test this).

Regarding converting this to a struct, we could structure it to something like:
```java
type Packages struct {
	PKGList  []string   `yaml:"packageList"`
	AddRepos []AddRepos `yaml:"additionalRepos"`
	RegCode  string     `yaml:"registrationCode"`
}

type AddRepos struct {
	Path       string `yaml:"path"`
	NoGPGCheck bool   `yaml:"noGPGCheck"`
}
```
We could add other properties as well such as: `auto-refresh`, or `priority`, but honestly it seems a bit like over engineering and IMO it might be better to add each property when the need occurs. The `noGPGCheck` makes sense since we might want to have a more granular `gpg-check` logic over the global `gpg-check` that will be added under the `Packages` struct. 

Does this seem enough to do at least for a first iteration, or did you have something different in mind and I have misunderstood?

Reply:
Taking a second stab at what I said previously regarding the path to local repos/.repo files. With the current implementation the user would have to specify the exact path to the dir as seen in the EIB in order for this to work. So it would have to be something like (assuming config dir is named `eib-dir`): 

```yaml
packages:
    additionalRepos:
    - /eib-dir/foo/bar
    - /eib-dir/foo/bar/foo.repo
```
In order to be more user-friendly we would have to sanitise the paths, but this should not be much work.

Reply:
As discussed offline, this will be handled in a follow-up PR.